File: electronics-sales/src/main/java/com/electronicssales/models/dtos/ProductDto.java
package com.electronicssales.models.dtos;

import java.util.Collection;

import javax.validation.constraints.NotBlank;

import com.electronicssales.entities.Paragraph;

import org.springframework.web.multipart.MultipartFile;

import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;

@Data
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ProductDto {

    long id;

    @NotBlank
    String productName;

    long price;

    Collection<Long> categoryIds;

    Long manufacturerId;

    Collection<Paragraph> paragraphs;

    Collection<ProductParameterDto> productParameters;

    MultipartFile[] images;

    Integer quantity;
    
}

File: electronics-sales-frontend/src/app/models/dtos/product.dto.ts
import { ProductParameterDto } from './product-parameter.dto';
import { ParagraphDto } from './paragraph.dto';
export interface ProductDto {

  id?: number;

  productName: string;

  price: number;

  categoryIds: number[];

  manufacturerId: number;

  paragraphs: ParagraphDto[];

  productParameters: ProductParameterDto[];

  images: File[];

}
